### DISCUSSION TITLE
🤔 발표자료에 넣어야 할 내용 논의하기

### DISCUSSION THREAD

**Discussion by sryung1225** [marketkarly-team/marketkarly-client#113] [General]
발표까지 d-2
발표자료 만들기까지 d-1 !!

발표 내용으로 들어가야 할 부분에 대해서 생각나는대로 적어볼까요??🤔
~~댓글로 남겨주시면 내용 최신화 하겠습니다!~~ 글 수정이 되네요???? 편하신 방법으로 남겨주세요!

***

- 작업자 별 작업 내용 간략 소개
  - 필요 시 작업 분류 과정 소개

- 작업 일정
  - week 1 / week2 분류 기준
  - week 1 내용
  - week 2 내용

- 작업 소통 도구
  - 디스코드 : 빠른 소통용
    - github 알림봇
  - 깃헙
    - discussions
    - wiki
 
 - 작업 과정
   - github-flow
   - pull-request (+어푸과정)(+코드리뷰)
 
- 각 페이지 소개
  - 페이지에 적용된 기능 (필요 시 gif 첨부 또는 직접 시연)
  - 페이지에 적용된 특징 (ex. 특별히 grid를 이용했습니다 촤하ㅏ~)
  - 작업 시 고충 & 해결과정
  
- 인사이트 도출
  - 작업하기 전에 어떠한 목표를 가지고 진행을 했는가?
  - 작업을 어떤식으로 계획을 구성했는가?
  - 작업을 진행하면서 막히거나 어려운 부분이 어떤 부분이었는가?
  - 문제점을 어떠한 방식으로 극복했는가?
  - 이번 프로젝트를 통해 깨달은 것과 스스로 발전했다고 생각되는 부분이 무엇인가?

**Comment by sryung1225** [comment]
인사이트 도출 - 성령

Q. 작업하기 전에 어떠한 목표를 가지고 진행을 했는가?
A. 마크업 파트에서 협업이 원활 할 수 있도록 기여하기. 맡은 스크립트 파트를 수행해내면서 개인 역량 발전시키기

Q. 작업을 어떤식으로 계획을 구성했는가?
A. 프로젝트 1-2일차 내에 마크업 작업 방식 정해서 초반 셋팅 / 1주차 마크업 작업 + 다른 분께 충분한 코드 리뷰 / 이후 1-2일 동안 페이지 조립 / 2주차 스크립트 작업 / 발표 하루 전 작업 종료 및 정리

Q. 작업을 진행하면서 막히거나 어려운 부분이 어떤 부분이었는가? 그 문제점을 어떠한 방식으로 극복했는가?
A1. 다른 분들의 코드를 어떤 식으로 리뷰 드려야 할 지, 과정에서 고민을 많이 하게 됐습니다. 원래는 PR을 이용해서 최대한 코멘트를 남기려고 했으나 시간 상 남길 여유도, 그걸 해결할 시간을 마련하기도 쉽지 않았습니다. 결국에는 제가 페이지 조립 작업을 진행하던 날에 제가 구조만 직접 수정 + 노션 활용해서 설명 + 스크럼 시간에 화면 공유하며 발표 하는 방식을 진행하게 됐는데 다행히 아주 나쁘지 않은 선택이였다고 되돌아봅니다.
A2. 심화된 스크립트, 특히 서버를 다룬 적이 없다보니 회원가입 스크립트 작업에서 시간을 많이 뺏기고 제대로 된 기여가 어려웠습니다. 멘토링을 통해 어느정도 과정은 익혔지만 실전과의 괴리가 느껴졌습니다. 결과적으로는 다른 담당자분이 주된 도움을 주시고, 제가 보조하는 쪽으로 진행하게 됐으며, 대신에 프로젝트 내에서 기여할 수 있는 다른 일을 찾아서 하며 극복해내갔습니다 (현재 진행중...)

Q. 이번 프로젝트를 통해 깨달은 것과 스스로 발전했다고 생각되는 부분이 무엇인가?
A1. 배우는 입장에서 프로젝트를 진행하는 것이 아닌, 처음으로 리드하는 입장에서 프로젝트를 진행하게 됐는데 어떤 식으로 소통하고 의견을 정리 해야 하는 지에 대해 알아갈 수 있었습니다.
A2. 내 것만 보고 만드는 것이 아닌, 다른 사람들의 코드를 보면서 이건 이렇게 해야하는구나 저렇게 하면 조금 더 좋겠구나 이해하고 배워갈 수 있었습니다. 작업에 직접 반영하지 않더라도 따라서 쳐보거나 내 방식대로 바꿔보는 연습을 따로 했는데 조금 더 적극적인 학습 방법을 갖게 됐구나 느꼈습니다.

**Comment by chaewon0128** [comment]
Q. 작업하기 전에 어떠한 목표를 가지고 진행을 했는가?
- 올바른 마크업 사용을 바탕으로 레이아웃을 구현을 하고,수업시간에 배운 자바스크립트를 많이 활용해 보고 싶었습니다.
 
Q. 작업을 진행하면서 막히거나 어려운 부분이 어떤 부분이었는가? 그 문제점을 어떠한 방식으로 극복했는가?
- 이번 프로젝트에서 캐러셀과 모달창 작업을 맡게 되었는데. 캐러셀 같은 경우 Swiper 라이브러리를 사용하면서 자체 내장 코드에 대한 이해가 부족하여 버튼 비활성화 같은 오류가 많이 발생했습니다. 초반에는 영문 사이트가 아직 낯설어 구글 검색에만 의존하여 오류를 해결했는데, 나중에 swiper 홈페이지 api 명세를 읽어보니 더 쉽게 이해가 가고 해결할 수 있는 범위가 넓어졌습니다. 
- 마크업을 작성하면서 접근성을 고려하는 것이 쉽지 않다라는 것을 많이 느끼게 되었습니다. 초반 레이아웃 작성시 충분히 고려하지 못해 이후에 추가로 수정하는데 들어가는 시간이 많았고, 머지시 충돌도 생겼습니다. 좀 더 꼼꼼하게 마크업 작업을 해야겠다는 생각이 들었습니다. 

Q. 이번 프로젝트를 통해 깨달은 것과 스스로 발전했다고 생각되는 부분이 무엇인가?
- 이번 프로젝트를 통해 git 과 github 사용이 많이 익숙해진 점이 가장 큰 성과인 것 같습니다. 특히 github에 다양한 사용방안을 알 수 있게 되어 다음 협업시에도 무리없이 사용할 수 있을 것 같습니다. 아쉬웠던 점은 자바스크립트를 많이 활용하고 있지 못하다라는 생각이 들었습니다. 간단한 이벤트 동작 위주로만 구현을 하게되고 validation 체크 나 심화된 코드 작성을 하는 것이 어렵다고 느꼈습니다. 프로젝트가 끝난 후에도 간단한 토이 프로젝트를 하며 코드 사용에 익숙해져야 겠다는생각이 들었습니다.

**Comment by jaesukpark77** [comment]
Q. 작업하기 전에 어떠한 목표를 가지고 진행을 했는가?
- 접근성을 고려하면서 마크업을 진행하고, RESTful API / Ajax 활용해 보는 것이 목표였습니다.
- 팀을 이끌고 프로젝트를 진행해 보는 것이 목표였습니다.

Q. 작업을 진행하면서 막히거나 어려운 부분이 어떤 부분이었는가? 그 문제점을 어떠한 방식으로 극복했는가?
- 접근성을 이전 작업에서는 고려를 해본 적이 없어 어떻게 접근을 해야 되는지 고민이 많았습니다. 구글링과 성령님의 조언을 통해 접근성에 대해 고민해 보고 최대한 적용해 보기 위해 노력하였습니다.
- Ajax을 이용해 로그인 기능을 구현하는 부분에서 내가 처음 구조를 짰던 방식이 제대로 적용되지 않았습니다. 결국 구조를 수정을 하였지만 내가 원하는 결과를 도출하지 못하였습니다. 이 과정에서 강사님들의 조언과 구글링을 통해 작업을 진행해 보았지만 프로젝트 기간이 끝나서 마무리를 하지 못하였습니다. 해당 부분은 프로젝트가 마감이 되어도 지속적인 고민을 해보고 나의 어떤 부분이 문제였는지 생각해 보고 마무리를 완벽하게 할 필요가 있을 듯싶습니다.

Q. 이번 프로젝트를 통해 깨달은 것과 스스로 발전했다고 생각되는 부분이 무엇인가?
- 처음에 시안을 받고 모든 기능을 기간 내에 완벽히 구현을 할 수 있을 줄 알았다. 하지만 작업을 진행하면서 신경 쓸 부분이 생각보다 많다는 것을 배울 수 있었습니다. 하나의 기능을 어떻게 하면 좋은 코드로 refactoring 시킬 수 있을지에 대한 고민을 계속 가질 수 있는 좋은 기회였습니다. 
- 또한 팀 프로젝트는 정말 팀원들의 합이 중요하다는 것을 배울 수 있었습니다. 결국 팀원끼리 맞지 않으면 해당 팀은 힘들 수 있다는 정보를 들었지만, 저희 팀은 정말 저를 믿고 잘 따라와 주셔서 제가 프로젝트 계획에 있어서 큰 힘이 되었습니다.
- 과거 '완성도와 기간 내에 프로젝트 마감' 중 어떤 부분이 중요하다고 생각되는지에 대한 질문을 받은 적이 있었는데 이번 프로젝트를 통해 완성도가 중요하다는 것을 느꼈습니다. 프로젝트 내에 마감이 되었어도 해당 작업 과정이 부실하면 결국 언젠가는 무너져내릴 수밖에 없을 것 같다는 생각을 했습니다. 반면 완성도를 생각하면서 조금 늦더라도 탄탄하게 작업을 하면 분명 나중에 큰 사건이 발생하지 않을 것이라는 확신이 생겼습니다.
- 이번 프로젝트를 통해 제가 어떤 부분이 부족하고 앞으로 어떠한 부분을 보완하고 발전해 나가야 할지 느낄 수 있었던 프로젝트였습니다. 부족한 부분은 보완하고 잘했던 부분은 계속 유지하면서 좋은 개발자로 성장하기 위해 노력할 것입니다.

**Comment by jeong-dev** [comment]
Q1. 작업하기 전에 어떠한 목표를 가지고 진행을 했는가?
- 자바스크립트 기능 구현도 좋지만, 처음 해보는 프로젝트인 만큼 협업 과정을 체득하고 팀원들과 원활히 소통하기 위해 노력하고자 했습니다.

Q2. 작업을 어떤식으로 계획을 구성했는가?
 - 프로젝트 1-2일차 내에 팀원들과 초반 initial setting 완료
 - 1주차 html과 css 작업 모두 완료
 - 2주차 바닐라 자바스크립트 작업 완료
 - 발표 하루 전 작업 종료 후 발표 자료 작성 및 정리

Q3. 작업을 진행하면서 막히거나 어려운 부분이 어떤 부분이었는가? 그 문제점을 어떠한 방식으로 극복했는가?
 - 깃을 사용하는 데 있어서 어려움이 많았던 것 같습니다. 어떤 식으로 작동을 하는지 혹은 내가 원하는 대로 제어하려면 어떻게 해야 할 지 등을 잘 몰라서 헤매기도 했지만, 팀원 분들이 하나하나 천천히 알려주셔서 적응할 수 있었습니다.
 - 아직 자바스크립트를 다루는 것이 미숙하기에, 어떤 식으로 코드를 짜야 원하는 동작을 구현할 수 있을지 갈피를 잡지 못해 힘들었습니다. 그렇기에 최대한 많은 시간을 구글링에 할애하여 제가 원하는 정보들을 얻었고 그것들을 이리저리 조합하여 몇 줄짜리 코드라도 짤 수 있게 되었습니다.

Q4. 이번 프로젝트를 통해 깨달은 것과 스스로 발전했다고 생각되는 부분이 무엇인가?
 - 프로젝트 시작 전에는 민폐를 끼치게 될까봐 사실 걱정을 많이 했는데, 뭐든 하다 보면 어떻게든 된다(?)는 것을 느끼며 스스로 조금이나마 자신감을 가지게 되었고, 이러한 부분에서 발전했다고 생각합니다.

**Comment by realldk** [comment]
- 작업하기 전에 어떠한 목표를 가지고 진행을 했는가?
    - js 비동기 통신을 이용해서 서버에 데이터를 주고 받는 것을 해보기
    - 프로젝트가 어떻게 진행되는지, 팀 협업은 어떻게 하는지 경험해보기
    - 내가 배웠던 것을 제대로 쓸 수 있는지 경험해보기
- 작업을 어떤식으로 계획을 구성했는가?
    - 1주차에 프로젝트 초기설정과 html, css 마무리, 유효성/접근성 점검
    - 2주차에 JS와 마무리 유효성/접근성 점검
- 작업을 진행하면서 막히거나 어려운 부분이 어떤 부분이었는가?
    - 배운 내용을 실제로 쓰려고 하니 어렴풋이 기억이 났고, 헷갈리는 것이 많아서 기록했던 것과 검색을 통해서 다시 한 번 찾아봐야 했습니다. 그리고 검색을 통해서 정보를 찾을 때도 딱 유효한 정보를 찾는 것이 쉽지 않았고, 연관된 내용이 여러가지이고 그것들을 조합해야 해결할 수 있는 문제들도 있었습니다. 때문에 생각했던 것보다 시간이 많이 소요되었습니다.
    - swiper를 처음 사용할 때 실제 swiper의 코드가 어떻게 구성되어 있는지 몰라서 어려웠습니다. tab키를 이용해서 접근했을 때 순서대로 이동, focus 표시, focus에서 자동재생 정지, 슬라이드 다음 요소와 슬라이드 내부페이지로 취사 선택해서 들어갈 수 있게 표시 등을 구현하고 싶었지만 demos에 나와 있는 예시를 따라 써보고 api를 읽어보다보니 계획했던 예상소요시간이 지났습니다.
- 문제점을 어떠한 방식으로 극복했는가?
    - 정보의 바다 속에서 우선순위를 정해서 중요하고 빨리해야되는 일들을 먼저하고, 못하는 것들은 따로 기록을 해놓고, 프로젝트가 끝나고 남는 시간에 하는걸로 정했습니다.
    - 팀원들과도 상의해보았지만 답이 안나와서 범선생님께 질문을 드렸고, swiper의 on()를 활용, tab index를 이용해서 함수 작성 등의 힌트를 얻어서 프로젝트가 끝난 후에 개인적으로 해볼 생각입니다.
- 이번 프로젝트를 통해 깨달은 것과 스스로 발전했다고 생각되는 부분이 무엇인가?
    - 배우는 것과 실제로 써보는 것은 다르다. 프로젝트를 하든, 개인 과제를 하든 실제로 많이 써봐야겠다고 생각이 들었습니다.
    - 프로젝트를 처음 해봤는데 같이 일할 때 의사소통을 자주, 정확하게 하는게 중요한 것 같습니다.
    - 접근성에 대해서 많이 생각하게 되었습니다. 장애가 있는 사람도, 장애가 없는 사람도 모두 생각해서 균형을 맞출 것, 그러면서 성능도 챙겨야 한다는 것을 알았습니다.
    - 수업에서 배운 내용이 전부는 아니지만 어느정도 배운 줄 알았는데 아직도 배울 게 많았습니다.
    - swiper의 api를 보면서 parameter, method, 생성자 함수를 어떻게 쓰는지 조금은 알게 되었습니다.
